lib/jsdom/utils.js

```diff
@@ -336,6 +348,246 @@ exports.parseContentType = function parseContentType(contentType) {
   return null;
 };
 
+exports.decodeString = function decodeString(buffer, contentType, defaultEncoding, detectMetaCharset) {
```

Comment:
Wow, it is amazing you implemented all this! Thank you!!

Can you move this and its helpers to a separate file, e.g. lib/jsdom/living/helpers/encoding.js? And use an options object for detectMetaCharset?